third/nmh/man/mhparam.man
.\"
.\" %nmhwarning%
.\" $Id: mhparam.man,v 1.1.1.1 1999-02-07 18:14:21 danw Exp $
.\"
.\" include the -mh macro file
.so %etcdir%/tmac.h
.\"
.TH MHPARAM %manext1% MH.6.8 [%nmhversion%]
.SH NAME
mhparam \- print nmh profile components
.SH SYNOPSIS
.in +.5i
.ti -.5i
mhparam
\%[components]
\%[-all]
\%[-component] \%[-nocomponent]
.br
\%[\-version]
\%[\-help]
.in -.5i
.SH DESCRIPTION
\fIMhparam\fR writes the value of the specified profile component to the
standard output separated by newlines.  If the profile component is not
present, the default value (or nothing if there is no default) is printed.

If the switch `\-component' is given, then the component name is displayed
along with the profile components value.  This can be disabled with the
switch `\-nocomponent'.

If more than one component is specified in the `components' list, then
the switch `\-component' is on by default.  If only one component is
specified, then the switch `\-nocomponent' is on by default.

If `\-all' is specified, then all components in the nmh profile are
displayed and other arguments are ignored.

Examples:

.nf
.ta \w'AliasFile:'u+2n
.in +.5i
% mhparam path
Mail

% mhparam mhlproc
%libdir%/mhl

% mhparam \-component path
Path: Mail

% mhparam AliasFile rmmproc
AliasFile: aliases
rmmproc: rmmproc

% mhparam \-nocomponent AliasFile rmmproc
aliases
rmmproc
.in -.5i
.fi

\fIMhparam\fR is also useful in back\-quoted operations:

.nf
.in +.5i
% fgrep cornell.edu `mhpath +`/`mhparam aliasfile`

.in -.5i
.fi
.Fi
^$HOME/\&.mh\(ruprofile~^The user profile
.Sa
mh-profile\|(5)
.De
`\-nocomponent' if only one component is specified
`\-component' if more than one component is specified
.Ds
`components' defaults to none
.Co
None
.En


doc/news
This file is intended to keep track of user-visible changes to the
release, for the construction of the Athena release notes.

Changes from release 8.0 to release 8.1:

	* Athena version numbers are now dotted triplets, e.g. 8.1.0
	  or 8.1.12.  Paths which include all or part of the Athena
	  version number now take the form "pathname-version",
	  e.g. "srvd-8.1" or "dash.menus-8.1"

	* In release 8.0, running "xmkmf" with the default path would
	  generate a Makefile which looked for its include files and
	  libraries in /mit/x11, which could be one of several
	  lockers.  Now it is preferrable to use the vendor X
	  libraries, so this bit of glue is going away.  There will be
	  no /usr/athena/bin/xmkmf any more, nor any symlinks in
	  /usr/athena pointing to the x11 locker.

	  Users can still do something like "add -f x11r5" and then
	  "xmkmf", and things should work, if the x11r5 locker is
	  built for their platform.

	* Athena bitmaps have moved to /usr/athena/include/X11/bitmaps.
	  Previously, they were in /usr/athena/lib/X11/bitmaps before
	  because of the /mit/x11 glue.

	* The symlink /bin/csh -> /bin/athena/tcsh has gone away.  The
	  update to 8.0 will fix /bin/csh references in /etc/passwd
	  and /etc/passwd.local to refer to /bin/athena/tcsh instead.

	* emacs18 has gone away.

	* AFS includes and libraries are installed in
	  /usr/afsws/include and /usr/afsws/lib.

	* wcl includes and libraries are installed in /usr/athena.

	* The basic krb5 headers and libraries are installed in
	  /usr/athena.

	* emacs has been upgraded from 19.30 to 19.34.  There should
	  be no significant user-visible changes.

	* RCS has been upgraded from version 4 to version 5.7.

	* x3270 has been upgraded from version 1.2 to version 3.1.1.3.

	* sysinfo is now available under Irix.

	* /bin/athena/tcsh now has Hesiod support enabled, which means
	  that "~lockername" will work in a pathname in most cases.
	  (It won't work for users with local passwd information who
	  don't have their lockers attached.)

	* For improved ease of debugging, many binaries are installed
	  with symbol tables, and compiled without optimization.
	  Therefore, the contents of /bin/athena and /usr/athena/bin
	  may be somewhat larger, and the binaries may consume more
	  CPU cycles, although it's unlikely that anyone will notice
	  the speed difference.


third/nmh/uip/mhparam.c

/*
 * mhparam.c -- print mh_profile values
 *
 * Originally contributed by
 * Jeffrey C Honig <Jeffrey_C_Honig@cornell.edu>
 *
 * $Id: mhparam.c,v 1.1.1.1 1999-02-07 18:14:14 danw Exp $
 */

#include <h/mh.h>

extern char *mhlibdir;
extern char *mhetcdir;

char *sbackup = BACKUP_PREFIX;
char *slink = LINK;

static struct swit switches[] = {
#define	COMPSW	  0
    { "components", 0 },
#define	NCOMPSW	  1
    { "nocomponents", 0 },
#define	ALLSW	  2
    { "all", 0 },
#define VERSIONSW 3
    { "version", 0 },
#define	HELPSW	  4
    { "help", 4 },
#define DEBUGSW   5
    { "debug", -5 },
    { NULL, 0 }
};

struct proc {
    char *p_name;
    char **p_field;
};

static struct proc procs [] = {
     { "context",       &context },
     { "mh-sequences",  &mh_seq },
     { "buildmimeproc", &buildmimeproc },
     { "faceproc",      &faceproc },
     { "fileproc",      &fileproc },
     { "foldprot",      &foldprot },
     { "incproc",       &incproc },
     { "installproc",   &installproc  },
     { "lproc",         &lproc },
     { "mailproc",      &mailproc },
     { "mhlproc",       &mhlproc },
     { "moreproc",      &moreproc },
     { "msgprot",       &msgprot },
     { "mshproc",       &mshproc },
     { "packproc",      &packproc },
     { "postproc",      &postproc },
     { "rmfproc",       &rmfproc },
     { "rmmproc",       &rmmproc },
     { "sendproc",      &sendproc },
     { "showmimeproc",  &showmimeproc },
     { "showproc",      &showproc },
     { "version",       &version_num },
     { "vmhproc",       &vmhproc },
     { "whatnowproc",   &whatnowproc },
     { "whomproc",      &whomproc },
     { "etcdir",        &mhetcdir },
     { "libdir",        &mhlibdir },
     { "sbackup",       &sbackup },
     { "link",          &slink },
     { NULL,            NULL },
};


/*
 * static prototypes
 */
static char *p_find(char *);


int
main(int argc, char **argv)
{
    int i, compp = 0, missed = 0;
    int all = 0, debug = 0;
    int components = -1;
    char *cp, buf[BUFSIZ], **argp;
    char **arguments, *comps[MAXARGS];

    invo_name = r1bindex (argv[0], '/');

    /* read user profile/context */
    context_read();

    arguments = getarguments (invo_name, argc, argv, 1);
    argp = arguments;

    while ((cp = *argp++)) {
	if (*cp == '-') {
	    switch (smatch (++cp, switches)) {
		case AMBIGSW: 
		    ambigsw (cp, switches);
		    done (1);
		case UNKWNSW: 
		    adios (NULL, "-%s unknown", cp);

		case HELPSW: 
		    snprintf (buf, sizeof(buf), "%s [profile-components] [switches]",
			invo_name);
		    print_help (buf, switches, 1);
		    done (1);
		case VERSIONSW:
		    print_version(invo_name);
		    done (1);

		case COMPSW:
		    components = 1;
		    break;
		case NCOMPSW:
		    components = 0;
		    break;

		case ALLSW:
		    all = 1;
		    break;

		case DEBUGSW:
		    debug = 1;
		    break;
	    }
	} else {
	    comps[compp++] = cp;
	}
    }

    if (all) {
        struct node *np;

	if (compp)
	    advise(NULL, "profile-components ignored with -all");

	if (components >= 0)
	    advise(NULL, "-%scomponents ignored with -all",
		   components ? "" : "no");

	/* print all entries in context/profile list */
	for (np = m_defs; np; np = np->n_next)
	    printf("%s: %s\n", np->n_name, np->n_field);

    } if (debug) {
	struct proc *ps;

	/*
	 * Print the current value of everything in
	 * procs array.  This will show their current
	 * value (as determined after context is read).
         */
	for (ps = procs; ps->p_name; ps++)
	    printf ("%s: %s\n", ps->p_name, *ps->p_field ? *ps->p_field : "");

    } else {
        if (components < 0)
	    components = compp > 1;

	for (i = 0; i < compp; i++)  {
	    register char *value;

	    value = context_find (comps[i]);
	    if (!value)
		value = p_find (comps[i]);
	    if (value) {
	        if (components)
		    printf("%s: ", comps[i]);

		printf("%s\n", value);
	    } else
	        missed++;
	}
    }
    
    done (missed);
}


static char *
p_find(char *str)
{
    struct proc *ps;

    for (ps = procs; ps->p_name; ps++)
	if (!strcasecmp (ps->p_name, str))
	    return (*ps->p_field);

    return NULL;
}
